src/db/connection.rs

```diff
@@ -23,12 +47,22 @@ impl From<Connection> for RequestContext {
 impl RequestContext {
     /// Get a DatabaseContext from this struct that can be used to connect
     /// to the db
-    pub fn database_context(
-        &self,
+    pub fn database_context<'a, 'b>(
+        &'b self,
+        resource: &'a str,
+        token: Option<Token>,
         action: ActionType,
         modifier: ActionModifier,
-    ) -> DatabaseContext {
-        return DatabaseContext::from(&self.conn, action, modifier);
+    ) -> Result<DatabaseContext<'a>>
+    where
+        'b: 'a,
+    {
+        DatabaseContext::from(&self.conn, resource, token, action, modifier)
+    }
+
+    pub fn db_context_anon(&self, action: ActionType, modifier: ActionModifier) -> DatabaseContext {
```

Comment:
This name takes me a little to process. How about `db_context_for_anon_user`? It's a bit of a mouthful but it's more clear, at least to me.

Reply:
Done!